src/routes/routes.py

```diff
@@ -0,0 +1,100 @@
+import heapq as hq
+
+import geopandas as gpd
+import networkx as nx
+from shapely.geometry import Point
+
+vertex = str
+"""A tuple containing the latitude and longitude of a point"""
+
+route = list[vertex]
+"""A list of coordinates representing a route"""
+
+
+# based on: https://www.geeksforgeeks.org/dijkstras-shortest-path-algorithm-greedy-algo-7/
+def route_to_safety(
+    origin_points: list[vertex], danger_zone: gpd.GeoDataFrame, G: nx.MultiDiGraph
+) -> list[route]:
+    """
+    Routes a list of origin points to the nearest safe location.
+
+    :param origin_points: A list of vertices given as str IDs
+    :param danger_zone: A GeoDataFrame containing the danger zone polygon(s).
+    :param G: A graph corresponding to the road network
+    :return: A list of routes where each route corresponds to the origin point at the same index.
+    """
+    routes = []
+
+    for origin in origin_points:
+        sptSet = dict((node, False) for node in list(G.nodes))
+        dist: list[tuple[float, str]] = [(float("inf"), node) for node in G.nodes]
+        node_priority = {node: float("inf") for node in G.nodes}
+        hq.heapify(dist)
+        predecessor: dict[str, str | None] = {
+            node: None for node in G.nodes
+        }  # To track shortest path
+
+        update_priority(dist, node_priority, origin, 0)
+
+        while dist:
+            priority, smallest_node = hq.heappop(dist)
+
+            if (
+                priority == node_priority[smallest_node]
+            ):  # Only use values that are not outdated
+                if not sptSet[smallest_node]:
+                    sptSet[smallest_node] = True
```

Comment:
I suspect that `sptSet` is redundant since the priority queue already ensures that nodes are processed optimally when dequeued.